### PR TITLE
Use bug report template with data validation

### PR DESCRIPTION
**Pull request by jonalmeida**
Github supports a more [useful templating system](https://docs.github.com/en/communities/using-templates-to-encourage-useful-issues-and-pull-requests/syntax-for-issue-forms) that can provide hints, descriptions, and simple data validation.

We can try to use this to increase the quality of our bug reports.

You use try this template at: [jonalmeida/test](https://github.com/jonalmeida/test/issues/new?assignees=&labels=%F0%9F%90%9E+bug%2Cneeds%3Atriage&template=---bug-report.yml&title=%5BBug%5D%3A+)
The output of a completed issue: https://github.com/jonalmeida/test/issues/1

A rendered version of the issue template:

### CODE REVIEW

**Review by gabrielluong — APPROVED** [review]

File: .github/ISSUE_TEMPLATE/---bug-report.yml [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+    validations:
+      required: true
+  - type: markdown
+    attributes:
+      value: |
+        # Device information
+  - type: input
+    attributes:
+      label: Device name
+      description: The name of the device manufactured.
```
Comment (eliserichards) [inline .github/ISSUE_TEMPLATE/---bug-report.yml:46]: Nit:
```suggestion
      description: The name of the device manufacturer.
```

File: .github/ISSUE_TEMPLATE/---bug-report.yml [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+      placeholder: Pixel 2
+    validations:
+      required: false
+  - type: input
+    attributes:
+      label: Android version
+      description: You can find the Android version information the About section of your device's system settings.
```
Comment (eliserichards) [inline .github/ISSUE_TEMPLATE/---bug-report.yml:53]: Nit: 
```suggestion
      description: You can find the Android version information in the About section of your device's system settings.
```

File: .github/ISSUE_TEMPLATE/---bug-report.yml [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+      description: |
+        Device logs or crash information can greatly aid in debugging. You can find some details here on how to [retrieve device logs or crash IDs][log].
+
+        [log]: https://github.com/mozilla-mobile/fenix/wiki/Logging-Crash-Information
+    validations:
+      required: false
+  - type: textarea
+    attributes:
+      label: Additional information
+      description: >
```
Comment (eliserichards) [inline .github/ISSUE_TEMPLATE/---bug-report.yml:86]: Should this be `|`?
```suggestion
      description: |
```

**Review by eliserichards — COMMENTED** [review]
This is excellent! I think the best part about this is the required fields. It will hopefully lead to more thoughtful/thorough bug reports.

**Review by Amejia481 — APPROVED** [review]

File: .github/ISSUE_TEMPLATE/---bug-report.yml (hunk 1 above)
Comment (jonalmeida) [reply .github/ISSUE_TEMPLATE/---bug-report.yml:46]: Manufacturer would be something like Google, Xiaomi, Samsung. Looking at some phone review sites, I think right term is "device model".

So maybe: "The name of the device model and manufacturer."

File: .github/ISSUE_TEMPLATE/---bug-report.yml (hunk 3 above)
Comment (jonalmeida) [reply .github/ISSUE_TEMPLATE/---bug-report.yml:86]: Either works, but good catch: let's be consistent.

**Comment by jonalmeida** [comment]
Since we've had a few eyes on it, I'm going to merge this and see how it holds up over the weekend issues that get filed. 🙂

**Pull request merged**